app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -64,12 +66,12 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(hawk_escape_id(feed)).properties['Machine Id']
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
-        os = os_for(feed)
+        os = os_for(hawk_escape_id(feed))
```

Comment:
@Jiri-Kremser , the easy way to test is to run hservices locally on a machine that has a `/etc/machine-id` (like Fedora, RHEL, CentOS) and check that the Agent has a value on the Machine ID field. Then, run this refresh parser and make sure that the same value is stored on the database.